engine/src/operators/GroupBy.cpp

```diff
@@ -207,91 +150,82 @@ void distributed_groupby_without_aggregations(
-	std::vector<gdf_column_cpp> & aggregation_inputs,
-	const std::vector<gdf_agg_op> & agg_ops,
-	std::vector<gdf_column_cpp> & group_by_output_columns,
-	std::vector<gdf_column_cpp> & aggrgation_output_columns,
-	const std::vector<std::string> & output_column_names) {
-	cudf::table keys = ral::utilities::create_table(group_by_columns);
-	cudf::table values = ral::utilities::create_table(aggregation_inputs);
+void aggregations_with_groupby(std::vector<CudfColumnView> const & group_by_columns,
+   std::vector<cudf::column_view> const & aggregation_inputs,
+   std::vector<cudf::experimental::aggregation::Kind> const & agg_ops,
+   std::vector< std::unique_ptr<cudf::column> > & group_by_output_columns,
+   std::vector< std::unique_ptr<cudf::column> > & aggregation_output_columns) {
 
-	std::vector<cudf::groupby::operators> ops(agg_ops.size());
-	std::transform(agg_ops.begin(), agg_ops.end(), ops.begin(), [&](const gdf_agg_op & op) {
-		return gdf_agg_op_to_groupby_operators(op);
-	});
+   std::vector<cudf::experimental::groupby::aggregation_request> requests;
+   requests.reserve(agg_ops.size());
 
-	cudf::groupby::hash::Options options(false);  // options define null behaviour to be SQL style
+   for(size_t i = 0 ; i < agg_ops.size() ; ++i){
+	   requests.push_back(cudf::experimental::groupby::aggregation_request {.values = aggregation_inputs[i]});
+	   requests[i].aggregations.reserve(agg_ops.size());
 
-	cudf::table group_by_output_table;
-	cudf::table aggrgation_output_table;
-	std::tie(group_by_output_table, aggrgation_output_table) = cudf::groupby::hash::groupby(keys,
-                                            values, ops, options);
-	
-	init_string_category_if_null(group_by_output_table);
-    init_string_category_if_null(aggrgation_output_table);
-
-	group_by_output_columns.resize(group_by_output_table.num_columns());
-	for(size_t i = 0; i < group_by_output_columns.size(); i++) {
-		// TODO percy cudf0.12 port to cudf::column
-//		group_by_output_columns[i].create_gdf_column(group_by_output_table.get_column(i));
-//		group_by_output_columns[i].set_name(group_by_columns[i].name());
-	}
+	   requests[i].aggregations.push_back(std::make_unique<cudf::experimental::aggregation>(agg_ops[i]));
 
-	aggrgation_output_columns.resize(aggrgation_output_table.num_columns());
-	for(size_t i = 0; i < aggrgation_output_columns.size(); i++) {
-		// TODO percy cudf0.12 port to cudf::column
-//		aggrgation_output_columns[i].create_gdf_column(aggrgation_output_table.get_column(i));
-//		aggrgation_output_columns[i].set_name(output_column_names[i]);
-	}
+	   cudf::table_view keys {{group_by_columns}};
+	   cudf::experimental::groupby::groupby obj(keys);
+	   
+	   std::pair<std::unique_ptr<cudf::experimental::table>, std::vector<cudf::experimental::groupby::aggregation_result>> output = obj.aggregate( requests );
+
+	   if (output.first->num_columns()>0) {
+		   std::vector< std::unique_ptr<cudf::column> > gby_cols = output.first->release();
```

Comment:
unless im somehow reading this wrong this is calling obj.aggregate(requests) every single time that it adds and aggregation op, so on somethign like select sum(a), sum(b) , sum(c) it would calcluate

sum(a) then throw away result
both sum(a) then sum(b) then throw away result
now all 3 
throw away all execpt for the first output